Reporting some linter errors

Dear Benjamin,

I just ran the excellent [`ruff` linter](https://github.com/charliermarsh/ruff) by @charliermarsh on the code base and wanted to report about the outcome. The identified spots may need corresponding adjustments.

### `pyproject.toml`
```
[tool.ruff]
line-length = 120
exclude = [
  "wetterdienst/ui/cli.py",
  "**/__init__.py",
]
```

### Install and invoke
```
pip install ruff
ruff --ignore=E501 --ignore=F821 wetterdienst
```

### Results
```
wetterdienst/core/scalar/values.py:241:9: F841 Local variable `conversion_factors` is assigned to but never used
wetterdienst/provider/eccc/observation/api.py:118:9: F841 Local variable `name` is assigned to but never used
wetterdienst/util/network.py:50:19: F841 Local variable `ttl_value` is assigned to but never used
```

With kind regards,
Andreas.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -4,6 +4,8 @@ Changelog
 Development
 ***********
 
+- Slightly adapt the conversion function to satisfy linter
+
 0.43.0 (05.09.2022)
 *******************
 

--- a/wetterdienst/core/scalar/values.py
+++ b/wetterdienst/core/scalar/values.py
@@ -220,18 +220,23 @@ class ScalarValuesCore(metaclass=ABCMeta):
         :return: pandas DataFrame with converted (SI) values
         """
         # TODO: THIS DOESNT CURRENTLY WORK WITH PREFIXED TIDY DATA
-        def _convert_values_to_si(series, parameter: Optional[str] = None) -> pd.Series:
+        def _convert_values_to_si(
+            series,
+            conv_factors: Dict[str, Tuple[Union[operator.add, operator.mul], float]],
+            parameter: Optional[str] = None,
+        ) -> pd.Series:
             """
             Helper function to apply conversion factors column wise to a pandas DataFrame
 
             :param series: pandas Series that should be converted
             :param parameter: optional parameter string used for tidy data
+            :param conv_factors: dictionary of prepared conversion factors and operators
             :return: converted pandas Series
             """
             if not parameter:
                 parameter = series.name
 
-            op, factor = conversion_factors.get(parameter, (None, None))
+            op, factor = conv_factors.get(parameter, (None, None))
 
             if not op or not factor:
                 return series
@@ -243,14 +248,16 @@ class ScalarValuesCore(metaclass=ABCMeta):
         if self.sr.tidy:
             data = []
             for par, group in df.groupby(Columns.PARAMETER.value, sort=False):
-                group[Columns.VALUE.value] = _convert_values_to_si(group[Columns.VALUE.value], par)
+                group[Columns.VALUE.value] = group[Columns.VALUE.value].apply(
+                    _convert_values_to_si, conv_factors=conversion_factors, parameter=par
+                )
                 data.append(group)
             try:
                 return pd.concat(data)
             except ValueError:
-                return pd.DataFrame()  # columns=self._meta_fields
+                return pd.DataFrame()
 
-        return df.apply(_convert_values_to_si, axis=0)
+        return df.apply(_convert_values_to_si, axis=0, conv_factors=conversion_factors)
 
     def _create_conversion_factors(self, dataset) -> Dict[str, Tuple[Union[operator.add, operator.mul], float]]:
         """

--- a/wetterdienst/provider/eccc/observation/api.py
+++ b/wetterdienst/provider/eccc/observation/api.py
@@ -115,17 +115,12 @@ class EcccObservationValues(ScalarValuesCore):
         """
         meta = self.sr.df[self.sr.df[Columns.STATION_ID.value] == station_id]
 
-        name, from_date, to_date = (
-            meta[
-                [
-                    Columns.NAME.value,
-                    Columns.FROM_DATE.value,
-                    Columns.TO_DATE.value,
-                ]
+        from_date, to_date = meta[
+            [
+                Columns.FROM_DATE.value,
+                Columns.TO_DATE.value,
             ]
-            .values.flatten()
-            .tolist()
-        )
+        ].values.flatten()
 
         # start and end year from station
         start_year = None if pd.isna(from_date) else from_date.year

--- a/wetterdienst/util/network.py
+++ b/wetterdienst/util/network.py
@@ -3,8 +3,9 @@
 # Distributed under the MIT License. See LICENSE for more info.
 import os
 from io import BytesIO
-from typing import List, Optional, Union
+from typing import Dict, List, Optional, Tuple, Union
 
+from fsspec import AbstractFileSystem
 from fsspec.implementations.cached import WholeFileCacheFileSystem
 from fsspec.implementations.http import HTTPFileSystem
 
@@ -17,10 +18,10 @@ class NetworkFilesystemManager:
     Manage multiple FSSPEC instances keyed by cache expiration time.
     """
 
-    filesystems = {}
+    filesystems: Dict[str, AbstractFileSystem] = {}
 
     @staticmethod
-    def resolve_ttl(ttl: Union[int, CacheExpiry]):
+    def resolve_ttl(ttl: Union[int, CacheExpiry]) -> Tuple[str, int]:
 
         ttl_name = ttl
         ttl_value = ttl
@@ -46,8 +47,8 @@ class NetworkFilesystemManager:
         cls.filesystems[key] = filesystem_effective
 
     @classmethod
-    def get(cls, ttl=CacheExpiry.NO_CACHE):
-        ttl_name, ttl_value = cls.resolve_ttl(ttl)
+    def get(cls, ttl=CacheExpiry.NO_CACHE) -> AbstractFileSystem:
+        ttl_name, _ = cls.resolve_ttl(ttl)
         key = f"ttl-{ttl_name}"
         if key not in cls.filesystems:
             cls.register(ttl=ttl)

